### PR TITLE
Additional features required for Obyte Zeus

### PR DESCRIPTION
**Pull request by Papabyte**
Changes proposed in this pull request:
- add isValidAddress function
- add new request "light/get_definition_chash"
- private key can be provided as buffer

### CODE REVIEW

File: src/api.json [hunk 1]
```diff
@@ -65,6 +65,12 @@
   "light/get_definition": {
     "params": "string"
   },
+  "light/get_definition_chash": {
+    "params": {
+    "address": "string",
+    "max_mci?": "number"
```
Comment (bonustrack) [inline src/api.json]: The key max_mci should not have `?`, the type are defined in the file `index.d.ts`
```suggestion
    "max_mci": "number"
```

**Comment by bonustrack** [comment]
Thank you for your contribution @Papabyte, i found some small issues:
- The lint test is failing, you can run the script "lint" to see the errors, if you run the script with "--fix" argument most of the errors will be fixed automatically. 
- The method "light/get_definition_chash" should also be added on this page https://github.com/obytescript/byteball.js/blob/master/index.d.ts#L193 with the name "getDefinitionChash"

**Comment by Papabyte** [comment]
Thanks for review.
These points are fixed in last commits.
I have a doubt about this: https://github.com/Papabyte/byteball.js/blob/master/index.d.ts#L11
I don't know how if a buffer should be typed as object.

File: index.d.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ declare namespace Byteball {
     address?: string;
     definition?: any[];
     path?: string;
+    private_key_buff?: object;
```
Comment (bonustrack) [inline index.d.ts]: Just "any" should be fine
```suggestion
    private_key_buff?: any;
```

Comment (bonustrack) [reply index.d.ts]: Also is it possible to change the param name to `privateKey` instead of `private_key_buff`?

Comment (Papabyte) [reply index.d.ts]: I corrected that.

**Comment by bonustrack** [comment]
Looking great, thank you for your contribution!

**Pull request merged**